### PR TITLE
Fix configuration and github actions bugs

### LINKED ISSUE #3: [Feature]: Add github action workflows
### What problem are you trying to solve?

I'd like this repo to utilize github actions.

### Describe the solution you'd like.

Add github workflows.

### What is the criteria for testers to pass this feature?

_No response_

### Describe alternatives you've considered.

_No response_

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by audrey-kanc**
## Description
This pull request makes significant updates to various configuration files, issue templates, and documentation to improve consistency, readability, and maintainability. Below are the most important changes grouped by theme:

### Issue Templates:
* Converted bug report and feature request templates from Markdown to YAML format for better structure and readability. (`.github/ISSUE_TEMPLATE/bug_report.yml`, `.github/ISSUE_TEMPLATE/feature_request.yml`) [[1]](diffhunk://#diff-637f7b97bba458badb691a1557c3d4648686292e948dbe3e8360564378b653efL1-R4) [[2]](diffhunk://#diff-c6b098646bf32c644234bec14c2675ea2a3d9c2dc2df450a25aa0e7ff02323cdL1-R10)

### Workflow Updates:
* Changed the cron schedule format in workflow files to use single quotes for consistency. (`.github/workflows/backup-repo.yml`, `.github/workflows/hcl-appscan-sast-scan.yml`) [[1]](diffhunk://#diff-7cbe46c2ec67f8a176f2e03e52c33c8882639f5721e0d1eba3c21b24a6c24195L5-R5) [[2]](diffhunk://#diff-e80aff488549cf63166bb98f89f29e231e3366c12acfc258591b519ed57d84a6L5-R5)
* Updated the HCL AppScan SAST Scan action to use version `v1.0.6`. (`.github/workflows/hcl-appscan-sast-scan.yml`)

### Documentation and Configuration:
* Added a new Prettier configuration file to standardize code formatting across the project. (`prettier.config.js`)
* Updated the README and CONTRIBUTING files to include new sections on getting started, scripts, and contribution guidelines. (`README.md`, `CONTRIBUTING.md`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L3-R41) [[2]](diffhunk://#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055L13-L22)
* Removed the recommended setup for contributing from the CONTRIBUTING file to streamline the document. (`CONTRIBUTING.md`)

### Miscellaneous:
* Added new words to the cspell configuration to improve spell checking. (`.vscode/cspell.json`)
* Added a new StyleCop configuration file to enforce coding standards. (`stylecop.json`)

These changes aim to enhance the project's overall structure, enforce coding standards, and improve the developer experience.

## Related Issue

closes #3

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] I have read the **README** document.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have followed the guidelines in the **CONTRIBUTING** document.
- [X] My code follows the code style of this project.
- [X] I have linted the code locally prior to submission.
- [ ] ~I have added tests to cover all of my changes.~
- [ ] ~All new and existing tests have been run locally and pass.~
- [ ] ~I have run the app locally and verified the changes work as expected.~

## Screenshots (if appropriate)

N/A

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-name: "🐛 Bug Report"
+name: '🐛 Bug Report'
 description: We broke something, again.
-labels: ["bug", "needs triage :exclamation:"]
-title: "[Bug]: <enter title here>"
+labels: ['bug', 'needs triage :exclamation:']
+title: '[Bug]: <enter title here>'
```
Comment (obryckim) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: I would get rid of `[Bug]` as we are now using labels for all of this.
I am planning on removing those prefixes from our app dev issues too.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
-name: "🚀 Feature request"
+name: '🚀 Feature request'
 description: Suggest a new feature / enhancement for this project
-labels: ["enhancement", "needs triage :exclamation:"]
-title: "[Feature]: <enter title here>"
+labels: ['enhancement', 'needs triage :exclamation:']
+title: '[Feature]: <enter title here>'
```
Comment (obryckim) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: I would get rid of `[Feature]` as we are now using labels for all of this.
I am planning on removing those prefixes from our app dev issues too.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
+- NVM
+- Node (20.x)
+- VS Code
+  - This includes all recommended VS Code extensions (included in /.vscode/extensions.json)
+- Dependencies
+  - `npm install`
+
+## Scripts
+
+This repository has a set of scripts which apply to all packages.
```
Comment (obryckim) [inline README.md]: This wording doesn't apply to you. This is not a monorepo w/ many packages.

```suggestion
The following scripts are available in this repository:
```

File: README.md [hunk 4]
```diff
@@ -1,13 +1,44 @@
...
+
+- `format:check`: run prettier and check if files are properly formatted
+  - e.g. `npm run format:check`
+  - NOTE: this will not change any files; it will notify you if any files need reformatting
+- `format`: format code using prettier
+  - e.g. `npm run format`
+  - NOTE: this will **make changes** to files
+  - NOTE: it is recommended to run this when there are no pending changes on the branch
+    - this way, you can see all changes made by prettier
+- `lint`: lint code (javascript, typescript, markdown, styles, etc.)
```
Comment (obryckim) [inline README.md]: Your lint script only lints markdown. You should remove JS/TS/STYLES/etc.

File: README.md [hunk 5]
```diff
@@ -28,10 +59,19 @@ This repo contains 2 types of issue templates:
...
 
+## Contributing
+
+To contribute to this repository, please see the [contribution guidelines](CONTRIBUTING.md).
+
 <!-- reference urls -->
 
-[gh-action-cvs]: github.com/plantemoran-ai-engineering/.github/actions/workflows/code-analysis.yml
-[gh-action-lint]: github.com/plantemoran-ai-engineering/.github/actions/workflows/linting.yml
+[developer-workstation-config-docs]: https://plantemoran-appdev.github.io/github-process-docs/workstation-setup/workstation-config/
```
Comment (obryckim) [inline README.md:67]: Alphabetize all of these

File: README.md [hunk 6]
```diff
@@ -28,10 +59,19 @@ This repo contains 2 types of issue templates:
...
+[gh-action--backup-repo]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/backup-repo.yml
+[gh-action-cvs]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/code-analysis.yml
+[gh-action-lint]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/linting.yml
 [gh-creating-default-comm-health-file]: https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file
-[img-gh-action-cvs-badge]: github.com/plantemoran-ai-engineering/.github/actions/workflows/code-analysis.yml/badge.svg
-[img-gh-action-lint-badge]: github.com/plantemoran-ai-engineering/.github/actions/workflows/linting.yml/badge.svg
+[gh-action--appscan-sast]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/hcl-appscan-sast-scan.yml
```
Comment (obryckim) [inline README.md]: Keep w/ a consistent naming pattern.
e.g. `gh-action--<action-name>`
(you have some w/ 1 dash and some w/ 2)

Here is how I am currently doing it:
```text
[gh-action--appscan-codesweep]
[gh-action--appscan-sast]
[gh-action--codeql]
[gh-action--linting]
etc.
```

File: README.md [hunk 7]
```diff
@@ -28,10 +59,19 @@ This repo contains 2 types of issue templates:
...
+[img--gh-action-badge--backup-repo]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/backup-repo.yml/badge.svg
+[img--gh-action-badge--appscan-sast]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/hcl-appscan-sast-scan.yml/badge.svg
+[img-gh-action-cvs-badge]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/code-analysis.yml/badge.svg
+[img-gh-action-lint-badge]: https://github.com/plantemoran-ai-engineering/.github/actions/workflows/linting.yml/badge.svg
```
Comment (obryckim) [inline README.md]: Same w/ your badges... stay consistent

Here is how I am currently doing it:
```text
[img--gh-action-badge--appscan-codesweep]
[img--gh-action-badge--appscan-sast]
[img--gh-action-badge--codeql]
[img--gh-action-badge--linting]
etc.
```

Also note how the suffix matches exactly w/ the suffix of the `gh-action--` ones above.

File: package.json [hunk 8]
```diff
@@ -17,9 +17,13 @@
     "url": "https://github.com/orgs/plantemoran-ai-engineering/.github/issues"
   },
   "scripts": {
-    "lint": "markdownlint **/*.md"
+    "format": "prettier . --write --log-level log",
+    "format:check": "prettier . --check",
+    "lint": "markdownlint **/*.md",
+    "lint:md": "markdownlint **/*.md"
```
Comment (obryckim) [inline package.json]: You don't need this.
Your `lint` script only does 1 thing (lint MD) and uses the same command.
```suggestion
```

File: stylecop.json
Comment (obryckim) [inline stylecop.json:1]: This is not a .net repo, so this file is not needed; it does nothing in this repo.

**Review by obryckim — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (audrey-kanc) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Updated.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (audrey-kanc) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: Updated.

File: stylecop.json
Comment (audrey-kanc) [reply stylecop.json:1]: Removed

File: README.md (hunk 4 above)
Comment (audrey-kanc) [reply README.md]: Updated

File: README.md (hunk 5 above)
Comment (audrey-kanc) [reply README.md:67]: Done

File: README.md (hunk 6 above)
Comment (audrey-kanc) [reply README.md]: Good catch! Fixed.

File: README.md (hunk 7 above)
Comment (audrey-kanc) [reply README.md]: Good catch! Fixed.

**Review by obryckim — APPROVED** [review]

**Review by obryckim — CHANGES_REQUESTED** [review]
Your linting workflow is failing;

File: package.json [hunk 9]
```diff
@@ -17,9 +17,12 @@
     "url": "https://github.com/orgs/plantemoran-ai-engineering/.github/issues"
   },
   "scripts": {
-    "lint": "markdownlint **/*.md"
+    "format": "prettier . --write --log-level log",
+    "format:check": "prettier . --check",
+    "lint": "markdownlint **/*.md",
```
Comment (obryckim) [inline package.json]: I think the comma at the end is causing a problem.
Be sure to run all of your scripts locally to ensure your package.json is good
(npm install, lint, format:check)

**Review by obryckim — CHANGES_REQUESTED** [review]

File: package.json (hunk 9 above)
Comment (audrey-kanc) [reply package.json]: I had accepted your suggestion to pull out linting on markdown, but I didn't pull the project back down and run all the scripts.

**Review by obryckim — APPROVED** [review]

**Pull request merged**